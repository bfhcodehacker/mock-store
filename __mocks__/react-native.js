jest.mock(
  '@react-native-vector-icons/ionicons',
  () => 'MockedIonicons',
);

jest.mock(
  'react-native-linear-gradient',
  () => 'MockedLinearGradient'
);

const ReactNative = jest.requireActual('react-native'); // Import the actual module to retain other functionalities

ReactNative.StyleSheet.create = (styles) => styles;
ReactNative.Platform = {
  OS: 'android',
  select: () => null
};

module.exports = ReactNative;