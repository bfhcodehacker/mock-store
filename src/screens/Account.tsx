import {
  Image,
  ScrollView,
  Text,
  TouchableOpacity,
  View
} from "react-native";
import { useEffect } from "react";
import { useAppDispatch, useAppSelector } from "../app/hooks";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { AccountStackParamList } from "../stacks/AccountStack";
import { getUserData } from "../reducers/accountData";
import { GradientWrapper } from "../components/GradientWrapper";
import { styles } from '../styles/screens/Account';
import { Address } from "../types/account";
import { Location } from "../components/Location";
import { signUserOut } from "../lib/authHelpers";

const defaultImage = require('../assets/images/defaultProfile.jpg');

export const AccountScreen: React.FC = () => {
  const navigation = useNavigation<NativeStackNavigationProp<AccountStackParamList>>();
  const dispatch = useAppDispatch();
  const theme = useAppSelector(state => state.theme);
  const isLoggedIn = useAppSelector(state => state.account.isLoggedIn);
  const account = useAppSelector(state => state.account);
  const { signInData, userData } = account;

  const goToSignIn = () => {
    navigation.navigate('SignIn');
  }

  useEffect(() => {
    if (!isLoggedIn) {
      goToSignIn();
    } else if (!account.userData && account.userDataStatus === 'idle') {
      dispatch(getUserData());
    }
  }, [isLoggedIn]);

  if (!account.isLoggedIn || !signInData) {
    return (
      <GradientWrapper>
        <TouchableOpacity
          onPress={goToSignIn}
          style={[styles.signInBtn, { backgroundColor: theme.secondaryColor}]}
        >
          <Text style={[styles.signInText, theme.primaryFont]}>Sign In</Text>
        </TouchableOpacity>
      </GradientWrapper>
    );
  }

  const profileImage = signInData.image ? { uri: signInData.image } : defaultImage;
  const email = signInData.email?.split('@');
  const addresses = [userData?.address];

  const renderAddress = (address?: Address) => {
    if (!address) {
      return;
    }

    return (
      <View
        style={[
          styles.addressContainer,
          { backgroundColor: theme.secondaryColor, borderColor: theme.primaryColor }
        ]}
        key={address.address}
      >
        <Text style={[styles.addressText, theme.secondaryFont]}>{address.address}</Text>
        <Text style={[styles.addressText, theme.secondaryFont]}>
          {address.city + ', ' + address.stateCode + ' ' + address.postalCode}
        </Text>
      </View>
    );
  }

  const signOut = () => {
    signUserOut(dispatch);
  }

  return (
    <GradientWrapper>
      <ScrollView style={styles.scrollview}>
        <View style={[styles.profileBox, { backgroundColor: theme.secondaryColor} ]}>
          <View style={styles.profileTop}>
            <Image source={profileImage} style={styles.profileImage} resizeMode='cover'/>
            <View style={styles.profileInfo}>
              <Text style={[styles.profileText, theme.secondaryFont]}>
                {signInData.firstName + ' ' + signInData.lastName}
              </Text>
              {email?.length && (
                <View>
                  <Text style={[styles.profileText, theme.secondaryFont]}>{email[0] + '@'}</Text>
                  <Text style={[styles.profileText, theme.secondaryFont]}>{email[1]}</Text>
                </View>
              )}
            </View>
          </View>
        </View>
        {addresses.length && (
          <View style={[styles.addressBox, { backgroundColor: theme.secondaryColor }]}>
            <Text style={[styles.addressTitle, theme.secondaryFont]}>Addresses</Text>
            {addresses.map(address => renderAddress(address))}
          </View>
        )}
        <Location />
        <TouchableOpacity
          onPress={signOut}
          style={[styles.signOutBtn, { backgroundColor: theme.secondaryColor}]}
        >
          <Text style={[styles.signOutText, theme.secondaryFont]}>Sign Out</Text>
        </TouchableOpacity>
      </ScrollView>
    </GradientWrapper>
  );
}           
