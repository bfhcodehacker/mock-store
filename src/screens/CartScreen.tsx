import { ScrollView, Text, TouchableOpacity, View } from 'react-native';
import { useEffect, useMemo, useState } from 'react';
import { useAppDispatch, useAppSelector } from '../app/hooks';
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { CartItem } from '../components/CartItem';
import {
  addToCart,
  removeFromCart,
  incrementCartQty,
  decrementCartQty
} from '../reducers/cartData';
import { CartProduct } from '../types/cartTypes/cartTypes';
import { Product } from '../types/product';
import { styles } from '../styles/screens/Cart';
import { calculateCartTotal } from '../lib/helpers';
import { CartStackParamList } from '../stacks/CartStack';
import { FeaturedProducts } from '../components/FeaturedProducts';
import { ATCModal } from '../components/ATCModal';

export const CartScreen = () => {
  const dispatch = useAppDispatch();
  const cartCount = useAppSelector(state => state.cartData.cartCount);
  const items = useAppSelector(state => state.cartData.items);
  const featuredProducts = useAppSelector((state) => state.storeData.cartProducts);
  const theme = useAppSelector(state => state.theme);
  const navigation = useNavigation<NativeStackNavigationProp<CartStackParamList>>();
  const [productAddedToCart, setProductAddedToCart] = useState<Product | null>(null);

  useEffect(() => {
    navigation.getParent()?.setOptions({
      tabBarBadge: cartCount ? cartCount : undefined,
    });
  }, [navigation, cartCount]);

  const cartTotal = useMemo(() => calculateCartTotal(items), [items]);

  const incrementCartItem = (id: number) => {
    dispatch(incrementCartQty({ productId: id }));
  };

  const decrementCartItem = (id: number) => {
    dispatch(decrementCartQty({ productId: id }));
  };

  const removeCartItem = (id: number) => {
    dispatch(removeFromCart({ productId: id }));
  };

  const renderCartItem = (item: CartProduct) => {
    return (
      <CartItem 
        item={item}
        key={item.title} 
        increment={incrementCartItem}
        decrement={decrementCartItem}
        remove={removeCartItem}    
      />
    );
  };

  const goToHomeTab = () => {
    navigation.navigate('Home' as any);
  };

  const addToCartPress = (product: Product) => () => {
    dispatch(addToCart({product, qty: 1}));
    setProductAddedToCart(product);
  };

  const closeATCModal = () => {
    setProductAddedToCart(null);
  }

  const navigateToProduct = (product: Product) => () => {
    navigation.navigate('Home' as any, {
      screen: 'ProductDisplay', 
      params: { productId: product.id, name: product.title }
    });
  };

  return (
    <ScrollView style={styles.main}>
      <View style={styles.innerContainer}>
        {items.length ? (
          <View>
            <View style={styles.cartHeader}>
              <Text style={[styles.cartCount, theme.secondaryFont]}>
                Item Count: {cartCount}
              </Text>
              <Text style={[styles.cartHeaderTotal, theme.secondaryFont]}>
                Total: ${cartTotal}
              </Text>
            </View>
            {items.map(renderCartItem)}
          </View>
        ) : (
          <View style={styles.emptyCart}>
            <Text style={styles.emptyCartText}>There are no items in the cart</Text>
            <TouchableOpacity
              onPress={goToHomeTab}
              style={[styles.continueShoppingBtn, { backgroundColor: theme.primaryColor}]}
            >
              <Text style={[styles.continueShoppingText, theme.primaryFont]}>
                Continue Shopping
              </Text>
            </TouchableOpacity>
          </View>
        )}
      </View>
      {!!featuredProducts.length && (
        <FeaturedProducts
          title={'You Might Also Like'}
          addToCart={addToCartPress}
          navigateToProduct={navigateToProduct}
          products={featuredProducts}
          titleStyle={{color: theme.primaryColor}}
          descStyle={{color: theme.primaryColor}}
          boxStyle={{borderColor: theme.primaryColor}}
        />
      )}
      {!!productAddedToCart && (
        <ATCModal onClose={closeATCModal} product={productAddedToCart} />
      )}
    </ScrollView>
  );
}