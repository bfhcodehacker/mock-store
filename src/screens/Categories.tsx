import { ScrollView, View } from 'react-native';
import { GradientWrapper } from '../components/GradientWrapper';
import { useAppSelector } from '../app/hooks';
import { FeaturedCategory } from '../components/FeaturedCategory';
import { Category } from '../types/category';
import { useNavigation } from '@react-navigation/native';
import { HomeStackParamList } from '../stacks/HomeStack';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';

import { styles } from '../styles/screens/Categories';

export const CategoriesScreen = () => {
  const navigation = useNavigation<NativeStackNavigationProp<HomeStackParamList>>();
  const categories = useAppSelector((state) => state.storeData.categories);

  const onPress = (category: Category) => () => {
    navigation.navigate('ProductIndex', { category });
  }

  return (
    <GradientWrapper>
      <ScrollView style={styles.scrollView}>
        <View style={styles.container}>
        {categories.map(category => {
          return <FeaturedCategory
            onPress={onPress}
            category={category}
            key={category.slug}
            boxStyle={styles.boxStyle}
          />
        })}
        </View>
      </ScrollView>
    </GradientWrapper>
  );
}