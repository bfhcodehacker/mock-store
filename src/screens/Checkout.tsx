import { Text, View } from 'react-native';
import { MainStyles } from '../styles/MainStyles';

export const CheckoutScreen = () => {
  return (
    <View style={MainStyles.main}>
      <Text>Checkout</Text>
    </View>
  );
}