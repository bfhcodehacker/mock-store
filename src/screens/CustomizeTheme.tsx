import { useEffect, useState } from "react";
import { ScrollView, Text, TouchableOpacity, View } from "react-native";
import { initialThemeState, setNewTheme, ThemeState } from "../reducers/themeData";
import { styles } from "../styles/screens/CustomizeTheme";
import { ThemePreview } from "../components/Theme/ThemePreview";
import { ThemeModal } from "../components/Theme/ThemeModal";
import { useAppDispatch } from "../app/hooks";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { HomeStackParamList } from "../stacks/HomeStack";
import { getThemes, saveNewTheme, updateSavedThemes } from "../lib/themeHelper";
import { ThemeEditOption } from '../components/Theme/ThemeEditOption';
import Ionicons from "@react-native-vector-icons/ionicons";

export const CustomizeTheme = () => {
  const navigation = useNavigation<NativeStackNavigationProp<HomeStackParamList>>();
  const dispatch = useAppDispatch();
  const [editOption, setEditOption] = useState('');
  const [showSavedThemes, setShowSavedThemes] = useState(false);
  const [updatedTheme, setUpdatedTheme] = useState<ThemeState>({...initialThemeState});
  const [savedThemes, setSavedThemes] = useState<ThemeState[]>();
  const { primaryColor, primaryEndColor, primaryFont, secondaryColor, secondaryFont } = updatedTheme;

  const fetchThemes = async () => {
    const saved = await getThemes();
    if (saved.length) {
      const newTheme = saved[0];
      setUpdatedTheme(newTheme);
      setSavedThemes([{...initialThemeState}, ...saved]);
    }
  }

  useEffect(() => {
    fetchThemes();
  }, []);

  const changeOption = (option: string) => () => {
    const value = updatedTheme[option as keyof ThemeState];
    if (value) {
      setEditOption(option);
    }
  }

  const toggleShowSavedThemes = () => {
    setShowSavedThemes(!showSavedThemes);
  }

  const useSavedTheme = (index: number) => () => {
    if (savedThemes?.[index]) {
      dispatch(setNewTheme({...savedThemes[index]}));
      setUpdatedTheme({...savedThemes[index]});
    }
  }

  const deleteTheme = (index: number) => () => {
    if (savedThemes) {
      const changedThemes = [...savedThemes];
      changedThemes.splice(index, 1);
      if (changedThemes) {
        setSavedThemes(changedThemes);
        setUpdatedTheme(changedThemes[0]);
        dispatch(setNewTheme({...changedThemes[0]}));
        if (changedThemes.length > 1) {
          updateSavedThemes(changedThemes);
        } else {
          updateSavedThemes([]);
        }
      }
    }
  }

  const saveTheme = () => {
    dispatch(setNewTheme(updatedTheme));
    saveNewTheme(updatedTheme);
    navigation.goBack();
  }

  const updateTheme = (data: any) => {
    const newTheme = { ...updatedTheme };
    newTheme[editOption as keyof ThemeState] = data;
    setUpdatedTheme(newTheme);
    closeModal();
  }

  const closeModal = () => {
    setEditOption('');
  }

  const closeScreen = () => {
    navigation.goBack();
  }

  return (
    <ScrollView style={styles.container} scrollEnabled={!editOption}>
      <Text style={styles.title}>Customize Your App!</Text>
      {(savedThemes && savedThemes?.length > 1) && (
        <View style={styles.savedThemesBox}>
          <TouchableOpacity onPress={toggleShowSavedThemes} style={styles.savedThemesHeader}>
            <Text style={styles.savedThemesTitle}>Saved Themes</Text>
            <Ionicons name={showSavedThemes ? 'caret-up' : 'caret-down'} size={20} />
          </TouchableOpacity>
          {showSavedThemes && (
            <View style={styles.savedThemeBox}>
              {savedThemes.map((theme: ThemeState, index: number) => {
                return (
                  <View style={styles.savedThemeRow} key={theme.primaryColor + index}>
                    <TouchableOpacity onPress={useSavedTheme(index)} style={styles.savedThemeLeft}>
                      <Text style={styles.savedThemeText}>Primary Color:</Text>
                      <View style={[styles.savedThemeColor, {backgroundColor: theme.primaryColor}]} />
                    </TouchableOpacity>
                    {index !== 0 && (
                      <TouchableOpacity onPress={deleteTheme(index)}>
                        <Ionicons name='trash' size={20} />
                      </TouchableOpacity>
                    )}
                  </View>
                );
              })}
            </View>
          )}
        </View>
      )}
      <View style={styles.topContainer}>
        <ThemeEditOption
          currentColor={primaryColor}
          editOption='primaryColor'
          changeOption={changeOption}
        />
        <ThemeEditOption
          currentColor={primaryEndColor}
          editOption='primaryEndColor'
          changeOption={changeOption}
        />
        <ThemeEditOption
          currentColor={primaryFont.color}
          editOption='primaryFont'
          currentFont={primaryFont}
          changeOption={changeOption}
        />
        <ThemeEditOption
          currentColor={secondaryColor}
          editOption='secondaryColor'
          changeOption={changeOption}
        />
        <ThemeEditOption
          currentColor={secondaryFont.color}
          editOption='secondaryFont'
          currentFont={secondaryFont}
          changeOption={changeOption}
        />
      </View>
      <View style={styles.previewContainer}>
        <ThemePreview
          primaryColor={primaryColor}
          primaryFont={primaryFont}
          primaryColorEnd={primaryEndColor}
          secondaryColor={secondaryColor}
          secondaryFont={secondaryFont}
        />
      </View>
      <TouchableOpacity onPress={saveTheme} style={styles.saveThemeBtn}>
        <Text style={styles.saveThemeBtnText}>Save Theme</Text>
      </TouchableOpacity>
      <TouchableOpacity onPress={closeScreen} style={styles.saveThemeBtn}>
        <Text style={styles.saveThemeBtnText}>Close</Text>
      </TouchableOpacity>
      {!!editOption && (
        <ThemeModal
          theme={updatedTheme}
          editOption={editOption}
          updateTheme={updateTheme}
          closeModal={closeModal}
        />
      )}
    </ScrollView>
  )
}