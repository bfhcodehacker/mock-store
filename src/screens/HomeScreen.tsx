import { ActivityIndicator, Image, ScrollView, Text, View } from 'react-native';
import { useEffect } from 'react';
import { GradientWrapper } from '../components/GradientWrapper';
import { HomeStyles } from '../styles/HomeStyles';
import { useAppDispatch, useAppSelector } from '../app/hooks';
import { fetchCategories, fetchProducts } from '../reducers/storeData';

import { FeatureCategories } from '../components/FeaturedCategories';
import { Category } from '../types/category';
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { HomeStackParamList } from '../stacks/HomeStack';
import { FeaturedProducts } from '../components/featuredProducts';
import { addToCart } from '../reducers/cartData';
import { Product } from '../types/product';

const titleBackground = require('../assets/images/home-title.jpeg');

export const HomeScreen = () => {
  const dispatch = useAppDispatch();
  const featuredCategories = useAppSelector((state) => state.storeData.featuredCategories);
  const featuredProducts = useAppSelector((state) => state.storeData.featuredProducts);
  const navigation = useNavigation<NativeStackNavigationProp<HomeStackParamList>>();

  useEffect(() => {
    if (!featuredCategories?.length) {
      dispatch(fetchCategories());
      dispatch(fetchProducts());
    }
  }, []);

  const onCategoryPress = (category: Category) => () => {
    navigation.navigate('ProductIndex', { category });
  }

  const onShopAllPress = () => {
    navigation.navigate('Categories');
  }

  const addToCartPress = (product: Product) => {
    dispatch(addToCart({product, qty: 1}));
  }

  return (
    <GradientWrapper>
      <ScrollView>
        <View style={HomeStyles.titleImageBox}>
          <Image source={titleBackground} resizeMode='contain' style={HomeStyles.titleImage} />
          <Text style={HomeStyles.titleText}>Mock App</Text>
        </View>
        {!!featuredCategories.length ? (
          <FeatureCategories
            categories={featuredCategories}
            onPress={onCategoryPress}
            onShopAllPress={onShopAllPress}
          />
        ) : (
          <ActivityIndicator />
        )}
        {!!featuredProducts.length && (
          <FeaturedProducts
            title={'Featured Products'}
            addToCart={addToCartPress}
            products={featuredProducts}
          />
        )}
      </ScrollView>
    </GradientWrapper>
  );
}