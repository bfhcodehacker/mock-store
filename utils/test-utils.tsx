import React, { JSX, PropsWithChildren } from 'react'
import { render } from '@testing-library/react-native'
import type { RenderOptions } from '@testing-library/react-native'
import { Provider } from 'react-redux'

import type { AppStore, RootState } from './testStore'
import { setupStore } from './testStore'

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>
  store?: AppStore
}

export const testStore = setupStore({});

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = testStore,
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

/*
use the function below to preload a store state when rendering for tests

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}
*/