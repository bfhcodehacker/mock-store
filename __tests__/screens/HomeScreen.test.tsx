import { screen } from '@testing-library/react-native';
import { HomeScreen } from '../../src/screens/HomeScreen';
import { renderWithProviders } from '../../utils/test-utils';

test('home screen renders', async () => {
  const expectedTitle = 'Mock App';
  renderWithProviders(<HomeScreen />);
  const title = await screen.findByTestId('home-title');
  expect(title).toHaveTextContent(expectedTitle);
  expect(screen.toJSON()).toMatchSnapshot();
});