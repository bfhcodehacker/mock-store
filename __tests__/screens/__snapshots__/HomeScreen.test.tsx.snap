// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home screen renders 1`] = `
<MockedLinearGradient
  colors={
    [
      "#00008B",
      "#6e70f3ff",
    ]
  }
  style={
    [
      {
        "height": "100%",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 30,
            "paddingHorizontal": 22,
            "paddingTop": 10,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontSize": 24,
              },
              {
                "color": "#fff",
                "fontFamily": "Inter-Regular",
              },
            ]
          }
        >
          Welcome!
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "elevation": 5,
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 16,
                },
                {
                  "color": "#00008B",
                  "fontFamily": "Inter-Regular",
                },
              ]
            }
          >
            Customize App
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "borderColor": "black",
            "borderRadius": 20,
            "borderWidth": 2,
            "elevation": 3,
            "height": 200,
            "marginHorizontal": "5%",
            "marginTop": 10,
            "overflow": "hidden",
            "width": "90%",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            {
              "testUri": "../../../src/assets/images/home-title.jpeg",
            }
          }
          style={
            {
              "height": "100%",
              "resizeMode": "cover",
              "width": "100%",
            }
          }
        />
        <Text
          style={
            {
              "fontFamily": "LavishyYours-Regular",
              "fontSize": 60,
              "paddingTop": "10%",
              "position": "absolute",
              "textAlign": "center",
              "width": "100%",
            }
          }
          testID="home-title"
        >
          Mock App
        </Text>
      </View>
      <ActivityIndicator />
    </View>
  </RCTScrollView>
</MockedLinearGradient>
`;
